.ci/Treecheck/Treecheck.jl

```diff
@@ -0,0 +1,187 @@
+    println("# BEGIN ALL_CHANGED_FILES")
+    println(str)
+    println("# END ALL_CHANGED_FILES")
+
+    elements = split(str) # Split on whitespace
+
+    package_list = Package[]
+
+    for element in elements
+        for (k, v) in registrytoml.dict["packages"]
+            package_uuid_str = k
+            package_name = v["name"]
+            package_relpath = v["path"]
+            package = Package(package_name, package_uuid_str)
+            if element == package_relpath*"/Package.toml"
+                @info "" package element package_relpath
+                push!(package_list, package)
+            end
+        end
+    end
+
+    num_changed_packages = length(package_list)
+
+    if num_changed_packages > 10
+        # This step prevents us from launching too many CI jobs (e.g. a hundred jobs)
+
+        msg = "Too many changed packages: $(num_changed_packages)"
+        error(msg)
+    end
+
+    dict = Dict()
+    dict["include"] = []
+
+    for package in package_list
+        matrix_entry = Dict(
+            "PACKAGE_NAME" => package.name,
+            "PACKAGE_UUID" => package.uuid_str,
+        )
+        push!(dict["include"], matrix_entry)
+    end
+
+    # Necessary to avoid erroring on empty matrix:
+    if isempty(dict["include"])
+        json_str = ""
+    else
+        json_str = JSON3.write(dict)
+    end
+
+    println(Base.stderr, json_str)
+
+    gha_set_output("matrix", json_str)
+
+    return json_str
+end
+
+
+
+function _treehash_to_commithash(treehash::AbstractString)
+    cmd_1 = `git log --pretty=raw`
+    cmd_2 = `grep -B 1 "$(treehash)"`
+    my_pipeline = pipeline(cmd_1, cmd_2)
+    str = strip(_tryread(my_pipeline))
+    r = r"commit ([A-Za-z0-9]*?)\n"
+    m = match(r, str)
+    commithash = strip(m[1])
+    return commithash
+end
+
+function _tryread(cmd::Base.AbstractCmd)
+    str = try
+        read(cmd, String)
+    catch ex
+        # If an exception is thrown, run the command again, but send stdout/stderr to the CI log
+        run(cmd)
+
+        rethrow()
+    end
+    return str
+end
+
+check(package_uuid_str::AbstractString) = check(RegistryToml(), package_uuid_str)
+function check(registrytoml::RegistryToml, package_uuid_str::AbstractString)
+    package_name = registrytoml.dict["packages"][package_uuid_str]["name"]
+    package_relpath = registrytoml.dict["packages"][package_uuid_str]["path"]
+    package_abspath = joinpath(registrytoml.root, package_relpath)
+    package_toml = joinpath(package_abspath, "Package.toml")
+    versions_toml = joinpath(package_abspath, "Versions.toml")
+    package_dict = TOML.parsefile(package_toml)
+    versions_dict = TOML.parsefile(versions_toml)
+    package_git_repo_url = package_dict["repo"]
+    @testset "Treecheck for package: $(package_name)" begin
+        @test !isempty(versions_dict)
+        mktempdir() do tmpdir
+            cd(tmpdir) do
+                run(`git clone "$(package_git_repo_url)" MYCLONEDIR`)
+                cd("MYCLONEDIR") do
```

Comment:
It seems easier to just clone into `tmpdir`.

Reply:
Implemented in https://github.com/JuliaRegistries/General/pull/136859/commits/9622aa35c68fcf8157bfabec50caafa107e8e007